frontend/src/hooks/useCollaborateData.tsx

```diff
@@ -39,6 +39,21 @@ interface IStaleNotificationOptions {
     afterSubmitAction: () => void;
 }
 
+const omitUnwantedData = (obj: any, field: string): any => {
+    if (Array.isArray(obj)) {
+        return obj.map(value => omitUnwantedData(value, field));
+    } else if (typeof obj === 'object' && obj !== null) {
+        const result: any = {};
```

Comment:
Any particular reason we're declaring the result object as const, as even though const Object and Arrays are mutable, but ideally should be declared as let/var when we're going to update/modify them as we do with other datatypes

Reply:
When you declare a variable with const, it means that the variable itself cannot be reassigned to a new value. However, if the variable refers to an object or array, you can still modify the properties or elements within that object or array. The const keyword doesn't make the object itself immutable, just the variable reference.

Using const is often encouraged when a variable is not going to be reassigned because it communicates the intent more clearly and helps to prevent unintentional reassignments that could lead to bugs.